e2e/app/modal/base-modal.ts

```diff
@@ -0,0 +1,92 @@
+import * as fp from 'lodash/fp';
+import {ElementHandle, Page} from 'puppeteer';
+import {getPropValue} from 'utils/element-utils';
+import {waitWhileLoading} from 'utils/waits-utils';
+import Container from 'app/container';
+import Button from 'app/element/button';
+import Checkbox from 'app/element/checkbox';
+import Textarea from 'app/element/textarea';
+import Textbox from 'app/element/textbox';
+import {LinkText} from 'app/text-labels';
+
+const defaultXpath = '//*[@id="popup-root"]//*[@role="dialog" and contains(@class, "after-open")]';
+
+export default abstract class BaseModal extends Container {
+
+   protected constructor(page: Page, xpath: string = defaultXpath) {
+      super(page, xpath);
+   }
+
+   /**
+    * Method to be implemented by children classes.
+    * Check whether current page has specified web elements.
+    */
+   abstract isLoaded(): Promise<boolean>
+
+   async waitForLoad(): Promise<this> {
+      await this.waitUntilVisible();
+      await waitWhileLoading(this.page);
+      await this.isLoaded();
+      return this;
```

Comment:
do we need to return this?

Reply:
Good catch. It was a mistake since the beginning. I will create a new PR to make the global correction. This change is out of scope for this PR.